src/Simplify.cpp

```diff
@@ -3908,6 +3924,57 @@ class Simplify : public IRMutator {
             IRMutator::visit(op);
         }
     }
+    template <typename T>
+    Expr hoist_slice_vector(Expr a, Expr b) {
+        debug(4) << "Trying to hoist slice vector:\n... a = " << a << "\n... b = " << b << "\n";
+
+        const Call *call_a = a.as<Call>();
+        const Call *call_b = b.as<Call>();
+
+        internal_assert(call_a && call_b &&
+                        call_a->is_intrinsic(Call::slice_vector) &&
+                        call_b->is_intrinsic(Call::slice_vector));
+
+        if (!equal(call_a->args[1], call_b->args[1]) ||
+            !equal(call_a->args[2], call_b->args[2])) {
+            debug(4) << " ...unable to hoist - slice vector arguments don't match\n";
+            return T::make(a, b);
```

Comment:
One way to fix this would be to go back to passing the expr to hoist_slice_vector, and then returning the argument when no change is made. Then, the callers of hoist_slice_vector should make sure to pass the original expression when appropriate (i.e. only make a new T to pass to hoist_slice_vector if a or b changed).